pkg/api/v1/pod/util.go

```diff
@@ -278,6 +278,17 @@ func GetExistingContainerStatus(statuses []v1.ContainerStatus, name string) v1.C
 	return status
 }
 
+// GetIndexOfContainerStatus gets the index of status of container "name" from "statuses",
+// returns -1 if container is not found.
+func GetIndexOfContainerStatus(statuses []v1.ContainerStatus, name string) int {
```

Comment:
Could we copy the interface of GetContainerStatus here and return a second param rather than -1 when things aren't found? Otherwise it would be relatively easy to accidentally write code that panics using this.

Reply:
We will need to return index value of -1 even if we added a bool for 'not-found'. I'm wondering if it is helpful to repeat the same information that a index value of -1 / 0+ already gives you by adding a bool.

I understand GetContainerStatus needs the bool because it returns empty v1.ContainerStatus if not found. If it were to return a pointer instead, it could have done away with the bool imho.

Reply:
In Go I'd generally expect `0, false` as a response here, rather than `-1, false`.